### PR TITLE
[SPARK-14596][SQL] Remove not used SqlNewHadoopRDD and some more unused imports

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

Old `HadoopFsRelation` API includes `buildInternalScan()` which uses `SqlNewHadoopRDD` in `ParquetRelation`.
Because now the old API is removed, `SqlNewHadoopRDD` is not used anymore.

So, this PR removes `SqlNewHadoopRDD` and several unused imports.

This was discussed in https://github.com/apache/spark/pull/12326.
## How was this patch tested?

Several related existing unit tests and `sbt scalastyle`.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @cloud-fan

**Comment by cloud-fan** [comment]
There is a `SqlNewHadoopRDDState`, we should rename it to `FileScanRDDState`

**Comment by HyukjinKwon** [comment]
Thanks! I just renamed.

**Comment by HyukjinKwon** [comment]
test this please

File: core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala [hunk 1]
```diff
@@ -220,8 +220,8 @@ class HadoopRDD[K, V](
 
       // Sets the thread local variable for the file's name
       split.inputSplit.value match {
-        case fs: FileSplit => SqlNewHadoopRDDState.setInputFileName(fs.getPath.toString)
-        case _ => SqlNewHadoopRDDState.unsetInputFileName()
+        case fs: FileSplit => FileScanRDDState.setInputFileName(fs.getPath.toString)
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala]: @liancheng @yhuai Do you still use `HadoopRDD` to read data source relation? If not, I think we don't need to update file name here anymore.

Comment (yhuai) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala]: I think it is used by HiveTableScan

Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala]: I see. This is used in https://github.com/apache/spark/blob/d6dc12ef0146ae409834c78737c116050961f350/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/InputFileName.scala during code generation.

Should I remove the change of file name?

Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala]: i see, how about renaming it to `InputFileNameHolder`?

File: core/src/main/scala/org/apache/spark/rdd/SqlNewHadoopRDDState.scala [hunk 2]
```diff
@@ -20,8 +20,7 @@ package org.apache.spark.rdd
 import org.apache.spark.unsafe.types.UTF8String
 
 /**
- * State for SqlNewHadoopRDD objects. This is split this way because of the package splits.
- * TODO: Move/Combine this with org.apache.spark.sql.datasources.SqlNewHadoopRDD
+ * State for FileScanRDD objects. This is split this way because of the package splits.
  */
 private[spark] object SqlNewHadoopRDDState {
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/rdd/SqlNewHadoopRDDState.scala]: `SqlNewHadoopRDDState` is definitely not a good name here, how about `InputFileNameHolder`?

File: core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala (hunk 1 above)
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala]: Sure. Thanks.

File: core/src/main/scala/org/apache/spark/rdd/SqlNewHadoopRDDState.scala (hunk 2 above)
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/rdd/SqlNewHadoopRDDState.scala]: Yes. Thanks!

**Comment by HyukjinKwon** [comment]
@cloud-fan The commits I just submitted includes the changes for MiMa tests and some comments.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks! merging to master!

**Pull request closed without merging**